meta-mender-core/recipes-bsp/u-boot/u-boot-mender-common.inc

```diff
@@ -196,7 +199,7 @@ EOF
         cat >> ${S}/mender_Kconfig_fragment <<EOF
 CONFIG_ENV_OFFSET=$HEX_MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_1
 CONFIG_ENV_OFFSET_REDUND=$HEX_MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_2
-CONFIG_SYS_MMC_ENV_PART=0
+CONFIG_SYS_MMC_ENV_PART=${MENDER_UBOOT_CONFIG_SYS_MMC_ENV_PART}
```

Comment:
I would recommend to make the appropriate changes to the [core patches](https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-bsp/u-boot/patches/0002-Generic-boot-code-for-Mender.patch#L99) to allow this, and specifically changing,

```
+# ifdef CONFIG_SYS_MMC_ENV_PART
+#  if CONFIG_SYS_MMC_ENV_PART != 0
+#   error CONFIG_SYS_MMC_ENV_PART need to be set to 0. Make sure that: 1) All the instructions at docs.mender.io/devices/integrating-with-u-boot have been followed. 2) All required layers are included in bblayers.conf, including any board specific layers such as meta-mender-<board>
+#  endif
+# else
+   /* Use MMC partition zero to select whole user area of memory card. */
+#  define CONFIG_SYS_MMC_ENV_PART  0
```

to

```
+# ifndef CONFIG_SYS_MMC_ENV_PART
+   /* Use MMC partition zero to select whole user area of memory card. */
+#  define CONFIG_SYS_MMC_ENV_PART  0
```

Reply:
I know you did a similar change in meta-mender-tegra, but might as well modify the "core" to allow this use-case without additional patching.

Reply:
Done with 1680058fece1f22ec9da67b82faed28a063e1ad7